spec/factories/plate_factories.rb

```diff
@@ -43,39 +68,56 @@
                                aliquot_count: outer_requests[i] ? 1 : 0
         end
       end
+      # Overide the purpose name
       purpose_name { 'example-purpose' }
+      # Overive the purpose uuid
       purpose_uuid { 'example-purpose-uuid' }
+      # The plate purpose
       purpose { create :v2_purpose, name: purpose_name, uuid: purpose_uuid }
+      # Sets up the pools on the plate. Used by outer_requests to work out which
+      # requests to build, and which pools (ie. submissions) to assign them to.
+      # eg pool_sizes [96] will result in a plate with one pool of 96 requests
+      #    pool_sizes [8,8,8] will result in three pools of 8 wells each.
       pool_sizes { [] }
       parents { [] }
       children { [] }
       descendants { [] }
+      # Array of pcr_cycles set up for each pool
       pool_prc_cycles { Array.new(pool_sizes.length, 10) }
+      # The state of the library request for each pool
       library_state { ['pending'] * pool_sizes.length }
       stock_plate { create :v2_stock_plate_for_plate, barcode_number: is_stock ? barcode_number : 2 }
       ancestors { [stock_plate] }
       transfer_targets { {} }
+      # Sets the plate size
       size { 96 }
       include_submissions { false }
+      # Array of states for individual wells, used to overide plate state for, eg. failed wells
       well_states { [state] * size }
-      custom_metadatum_collection { nil }
       is_stock { false }
     end
 
     sequence(:id) { |i| i }
-    uuid { SecureRandom.uuid }
+    uuid
+    # Number of rows is calculated from the size by default
     number_of_rows { (((size / 6)**0.5) * 2).floor }
+    # Number of columns is calculated from the size by default
     number_of_columns { (((size / 6)**0.5) * 3).floor }
+    # The state of the plate. Can overide for individual wells using
+    # well_states
     state { 'pending' }
     created_at { '2017-06-29T09:31:59.000+01:00' }
     updated_at { '2017-06-29T09:31:59.000+01:00' }
+    custom_metadatum_collection { nil }
 
     # Mock the relationships. Should probably handle this all a bit differently
     after(:build) do |plate, evaluator|
-      RSpec::Mocks.allow_message(plate, :wells).and_return(evaluator.wells)
-      RSpec::Mocks.allow_message(plate, :purpose).and_return(evaluator.purpose)
-      RSpec::Mocks.allow_message(plate, :custom_metadatum_collection).and_return(evaluator.custom_metadatum_collection)
+      plate._cached_relationship(:wells) { evaluator.wells }
+      plate._cached_relationship(:purpose) { evaluator.purpose }
+      plate._cached_relationship(:custom_metadatum_collection) { evaluator.custom_metadatum_collection }
```

Comment:
How does this _cached_relationship work?
Also, is the `# Should probably handle this all a bit differently` comment still accurate after making this change?

Reply:
Good question, I'll pop a description in the code, as it will likely cause similar questions in future.
The # Should probably handle this all a bit differently is still true, although less so now, and the alternatives have their own issues.

Reply:
Explanation added